src/integration-test/java/com/commercetools/sync/integration/commons/utils/ITUtils.java

```diff
@@ -65,21 +64,30 @@ public static String getStatisticsAsJSONString(@Nonnull final BaseSyncStatistics
      * @param resourceMapper       defines a mapper function that should be applied on each resource in the fetched page
      *                             from the query on the specified CTP project.
      */
-    public static <T extends Resource, C extends QueryDsl<T, C>> void queryAndApply(
+    public static <T extends Resource, C extends QueryDsl<T, C>> void queryAndExecute(
         @Nonnull final SphereClient ctpClient,
         @Nonnull final Supplier<QueryDsl<T, C>> queryRequestSupplier,
         @Nonnull final Function<T, SphereRequest<T>> resourceMapper) {
+        queryAll(ctpClient, queryRequestSupplier.get(), resourceMapper)
+            .thenApply(sphereRequests -> sphereRequests.stream()
+                                                       .map(ctpClient::execute)
```

Comment:
could `queryAndCompose` be reused for this method implementation?

Reply:
thanks: c9c983b